src/System.Linq/src/System/Linq/Distinct.cs

```diff
@@ -116,6 +116,11 @@ public List<TSource> ToList()
                 return FillSet().ToList();
             }
 
+            public HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                return FillSet().ToHashSet(comparer);
```

Comment:
Instead of `Equals` use [`Utilities.AreEqualityComparersEqual(comparer, _comparer)`](https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/Utilities.cs#L21) should probably be used. (Also @JonHanna you forgot to pass the comparer to the set in your comment.)